File: ts/problems/58-length-of-last-word.ts
export function lengthOfLastWord(s: string): number {
  // return s.trimEnd().split(" ").pop()?.length || 0;
  if (!s.length) return 0;
  let length = 0;
  for (let i = s.length - 1; i >= 0; i--) {
    if (s[i] !== " ") length++;
    if (length && s[i] === " ") return length;
  }
  return length;
}

File: go/problems/58_length_of_last_word.go
package main

func lengthOfLastWord(s string) int {
	// if len(s) == 0 {
	// 	return 0
	// }
	//
	// var length int
	// for i := len(s) - 1; i >= 0; i-- {
	// 	if string(s[i]) != " " {
	// 		length++
	// 		continue
	// 	}
	// 	if length > 0 && string(s[i]) == " " {
	// 		return length
	// 	}
	// }
	//
	// return length

	end := len(s) - 1
	for end >= 0 && s[end] == ' ' {
		end--
	}

	if end < 0 {
		return 0
	}

	start := end
	for start >= 0 && s[start] != ' ' {
		start--
	}

	return end - start
}
